test/Hamiltonians.jl

```diff
@@ -77,6 +77,8 @@ end
         GutzwillerSampling(HubbardReal1D(BoseFS((1,2,3)); u=6); g=0.3),
         GutzwillerSampling(BoseHubbardMom1D2C(BoseFS2C((3,2,1), (1,2,3)); ua=6); g=0.3),
         GutzwillerSampling(HubbardReal1D(BoseFS((1,2,3)); u=6 + 2im); g=0.3),
+
+        # MatrixHamiltonian([1 2;2 0]),
```

Comment:
This test fails currently because it expects the `starting_address` to be an `AbstractFockAddress` and `num_modes()` to be defined for it. I think this conceptually wrong.